sound/soc/sof/ipc.c

```diff
@@ -639,90 +639,60 @@ EXPORT_SYMBOL(snd_sof_ipc_stream_posn);
 /*
  * IPC get()/set() for kcontrols.
  */
-
-int snd_sof_ipc_set_comp_data(struct snd_sof_ipc *ipc,
-			      struct snd_sof_control *scontrol, u32 ipc_cmd,
-			      enum sof_ipc_ctrl_type ctrl_type,
-			      enum sof_ipc_ctrl_cmd ctrl_cmd)
+int snd_sof_ipc_set_get_comp_data(struct snd_sof_ipc *ipc,
+				  struct snd_sof_control *scontrol,
+				  u32 ipc_cmd,
+				  enum sof_ipc_ctrl_type ctrl_type,
+				  enum sof_ipc_ctrl_cmd ctrl_cmd,
+				  bool send)
 {
 	struct snd_sof_dev *sdev = ipc->sdev;
 	struct sof_ipc_ctrl_data *cdata = scontrol->control_data;
-	int err;
+	size_t send_bytes;
+	int err = 0;
 
-	/* read firmware volume */
+	/* read or write firmware volume */
```

Comment:
is this actually just for volume? Wondering if the comment is still relevant?

Reply:
really don't know when this is used. As this is legacy code I just left it as it was...

Reply:
let's ask @ranj063